src/modules/checkout/actions.ts

```diff
@@ -105,7 +105,15 @@ export async function submitDiscountForm(
 }
 
 export async function setAddresses(currentState: unknown, formData: FormData) {
-  if (!formData) return "No form data received"
+  const headersList = headers();
+  const referer = headersList.get('referer');
```

Comment:
I used referer here to extract the lang and locale parameters from the url. I have tried to use the `window.location.href`, but it seems like since we are in the server actions, the `window` is undefined.